### PR TITLE
Update tutorial for adding siteMetadata

### PR DESCRIPTION
**Pull request by moxiegirl**
- The tutorial is adding a graphql query to `about.js` which appears to be an artifact from v1 tutorial. This throws a warning on compile. 
- The StaticQuery in the layout goes into each file. This is super useful, added explanation for that.
- Also, I found the discourse hard to follow, specific references to files and so forth can help

### CODE REVIEW

**Comment by KyleAMathews** [comment]
Deploy preview for *using-postcss-sass* failed.

Built with commit f9bf0beee41ef5538164502800e07b3d649428a0

https://app.netlify.com/sites/using-postcss-sass/deploys/5b608202c9659273c23ff6e4

**Comment by KyleAMathews** [comment]
Deploy preview for *using-postcss-sass* failed.

Built with commit a0cd53a2a559d65b7945066bb431909b4ee982db

https://app.netlify.com/sites/using-postcss-sass/deploys/5b6082721f12b7454e6546b9

**Comment by amberleyromo** [comment]
Hey @moxiegirl! Responses inline below:

> The tutorial is adding a graphql query to about.js which appears to be an artifact from v1 tutorial. This throws a warning on compile.

The `graphql` page query is still valid in v2 -- the `StaticQuery` API is new in v2. See [this section of the docs](https://next.gatsbyjs.org/docs/static-query/#how-staticquery-differs-from-page-query) for the difference between `StaticQuery` and `graphql` page query. (Page queries only work in `src/pages/` files, but you can use `StaticQuery` everywhere).

I went through the existing tutorial steps and did not get a warning on compile. (I did notice one of the line changes (the `/>` closing bracket of the `<StaticQuery>` isn't highlighted and should be -- that definitely needs to be adjusted, and perhaps caused your compile error?)

Can you look again at following the steps in the existing tutorial? If you still get that compile error, can you throw it up here? If not, do you want to make that change to the line highlighting for that closing bracket, instead?

> The StaticQuery in the layout goes into each file. This is super useful, added explanation for that.

Not sure what you mean "each file"? These PR changes show the Static Query in `src/components/layout.js` but not the `src/pages/about.js`. The two main changes I see in the PR: 
1) remove the `graphql` import from the `about.js` page (but maintain the `graphql` query -- this would break it), and
2) change the order of editing the `about.js`/`layout.js` files (switched). Also the highlighting of the line edits appears mismatched now. 

It doesn't look to me like these changes to the code examples are needed.

> Also, I found the discourse hard to follow, specific references to files and so forth can help

Thanks for the feedback on the explanations -- +1 on some of your content suggestions!


Let me know if you take another look and still get those errors. Otherwise can you edit this PR to reflect just the content changes and we'll go from there?

**Comment by moxiegirl** [comment]
@amberleyromo Sure. I'll take a look.

As I recall, one `.js` was using the old query method and the other the new.  Shouldn't they both be using the new method?  

"each file" what I meant is that by virtue of the query being in the layout and file that uses that layout also has access to the query data.  I'll clarify this.  I believe I switched the order so the query is added to the layout first and the files continue to work...I'll double check.

**Comment by moxiegirl** [comment]
>  went through the existing tutorial steps and did not get a warning on compile. (I did notice one of the line changes (the /> closing bracket of the <StaticQuery> isn't highlighted and should be -- that definitely needs to be adjusted, and perhaps caused your compile error?)

Yes,  I went through it again. The order of changes in the page is edit `about.js` to use the global site data and _then_ add the query to `layout.js`.  See here:

image

If the reader is running in develop mode and saving each file as it changes, this order resutls in an error between adding the data reference in the `about.js` file and adding the query that supplies the data in `layout.js`.  The order appears in both the console and the site. That is why I reversed the order. The error I get makes sense of course:

```bash
✖ ｢wdm｣: 
ERROR in ./src/pages/about.js
Module Error (from ./node_modules/eslint-loader/index.js):

/Users/moxiegirl/repos/fourtest/src/pages/about.js
  6:20  error  'data' is not defined  no-undef

✖ 1 problem (1 error, 0 warnings)

 @ ./.cache/sync-requires.js 14:50-111
 @ ./.cache/app.js
 @ multi ./node_modules/react-hot-loader/patch.js (webpack)-hot-middleware/client.js?path=http://localhost:8000/__webpack_hmr&reload=true&overlay=false ./.cache/app
ℹ ｢wdm｣: Failed to compile.
```

I prefer not to force the reader into an error situation at this point. It can be instructive but is perhaps not the best thing to teach initially.

**Comment by KyleAMathews** [comment]
Deploy preview for *using-jss* failed.

Built with commit da0666b548a8b5af86fd16e269e58acdf2a176f0

https://app.netlify.com/sites/using-jss/deploys/5b67c2f482d3f14212bdce17

**Comment by moxiegirl** [comment]
Ok, I left the order as is but added explanation to clarify two techniques. It provides more context as to what is being done.  And why. I think I'm done for now.  Thanks for the help @amberleyromo ...

File: docs/tutorial/part-four/index.md [hunk 1]
```diff
@@ -242,39 +243,44 @@ module.exports = {
...
-)
+In your page, you would add a query inside a `<StaticQuery/>` tag somewhere in
+your source code, for example:
 
-export const query = graphql`
-  query {
-    site {
-      siteMetadata {
-        title
+```jsx
```
Comment (amberleyromo) [inline docs/tutorial/part-four/index.md]: I would remove this section, and leave the example usage just in the code snippet for the `layout.js` change.

File: docs/tutorial/part-four/index.md [hunk 2]
```diff
@@ -242,39 +243,44 @@ module.exports = {
 }
 ```
 
-Restart the development server.
+Restart the development server. Now, you can access the `title` with a GraphyQl
```
[the remark this answers is not in the document]
Comment (moxiegirl) [reply docs/tutorial/part-four/index.md]: Sure thing.  If the thinking is that the `StaticQuery` is an outlier or less prevalent than the page query. I'll make that change.

File: docs/tutorial/part-four/index.md [hunk 3]
```diff
@@ -208,22 +208,23 @@ You have another small site with a layout and two pages.
 
 Now let's start querying 😋
 
-## Querying for the site title
+## Add and query for siteMetadata
 
-When building sites, it's common to want to reuse common bits of data across the
+When building sites, designers often to want to reuse common bits of data across the
```
[the remark this answers is not in the document]
Comment (moxiegirl) [reply docs/tutorial/part-four/index.md]: 👍

File: docs/tutorial/part-four/index.md (hunk 1 above)
Comment (moxiegirl) [reply docs/tutorial/part-four/index.md]: The reason I put that in is that the section is all about GraphQL but we don't actually show a simple query. Moreover, the simple query syntax is buried in the format of the page/StaticQuery.  When readers are moving fast, they often miss the fine points...For example, I only recall one mention somewhere that users can test queries out in `http://localhost:8000/___graphql` --- this seems like a good point to illustrate it again.

Comment (amberleyromo) [reply docs/tutorial/part-four/index.md]: I agree -- my comment was about the `... source code ...` Static Query snippet below. Id take that out and just show the usage in `layout.js` -- not include this extra section.

Comment (moxiegirl) [reply docs/tutorial/part-four/index.md]: Ah, gotcha, I think I've got it right.

**Comment by moxiegirl** [comment]
Thanks, @amberleyromo for all your help.  I've made the changes, rebased against master, and pushed again. I couldn't resist switching out the Bret Victor paragraph with the discussion of `How does the graphql tag work?` --- shouldn't the how does it work immediately follow the use?

**Comment by KyleAMathews** [comment]
Deploy preview for *using-glamor* failed.

Built with commit 39dfe1b7a281f736f84df99e738b7b9d35fe4890

https://app.netlify.com/sites/using-glamor/deploys/5b6cd0a21f12b71dd3ac780d

File: docs/tutorial/part-four/index.md [hunk 4]
```diff
@@ -274,7 +272,33 @@ export const query = graphql`
 `
 ```
 
-`src/components/layout.js`
+It worked!! 🎉
+
+![fake-title-graphql](fake-title-graphql.png)
+
+Page queries must live outside of the source code, by convention at the end of a
```
Comment (amberleyromo) [inline docs/tutorial/part-four/index.md]: "outside of the source code" => "outside of the component definition"

"by convention at the end of a page" => "by convention at the end of the page component file"

just to clarify a bit more

File: docs/tutorial/part-four/index.md [hunk 5]
```diff
@@ -274,7 +272,33 @@ export const query = graphql`
...
+page. Your new code retrieves the `title` with this GraphyQl query:
```
Comment (amberleyromo) [inline docs/tutorial/part-four/index.md]: "GraphyQl" => "GraphQL"

File: docs/tutorial/part-four/index.md (hunk 4 above)
Comment (moxiegirl) [reply docs/tutorial/part-four/index.md]: 👍

File: docs/tutorial/part-four/index.md (hunk 5 above)
Comment (moxiegirl) [reply docs/tutorial/part-four/index.md]: 👍

File: docs/tutorial/part-four/index.md [hunk 6]
```diff
@@ -66,7 +66,7 @@ to you.
 
 Gatsby uses GraphQL to enable components to declare the data they need.
 
-## Your first GraphQL query
+## Create a new example site
```
Comment (m-allanson) [inline docs/tutorial/part-four/index.md]: Changing this will break the "Your first GraphQL query" link in the sidebar:

screen shot 2018-08-10 at 18 52 08

I think "Your first GraphQL query" works well for the sidebar, so maybe it makes sense to keep it here? Or change it to something else. Anyway I think this is good to merge, and the sidebar can be updated in a follow-up PR.
